Special characters in metadata fields cause Gunicorn to 500

<!--
    NOTE: This issue should be for problems with PyPI itself, including:
    * pypi.org
    * test.pypi.org
    * files.pythonhosted.org

    This issue should NOT be for a project installed from PyPI. If you are
    having an issue with a specific package, you should reach out to the
    maintainers of that project directly instead.

    Furthermore, this issue should NOT be for any non-PyPI properties (like
    python.org, docs.python.org, etc.)
-->

**Describe the bug**
I want to publish a new package (new package, not only new version).
Twine comes back with a 500 server error.

**Expected behavior**
Package should be registered and uploaded.

**To Reproduce**

```bash
$ twine upload dist/*
Uploading distributions to https://upload.pypi.org/legacy/
Uploading django_dynamic_filenames-1.0.0-py3-none-any.whl
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10.3k/10.3k [00:01<00:00, 7.11kB/s]
Received "500: Internal Server Error" Package upload appears to have failed.  Retry 1 of 5
Uploading django_dynamic_filenames-1.0.0-py3-none-any.whl
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10.3k/10.3k [00:00<00:00, 26.4kB/s]
Received "500: Internal Server Error" Package upload appears to have failed.  Retry 2 of 5
Uploading django_dynamic_filenames-1.0.0-py3-none-any.whl
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10.3k/10.3k [00:00<00:00, 20.8kB/s]
Received "500: Internal Server Error" Package upload appears to have failed.  Retry 3 of 5
Uploading django_dynamic_filenames-1.0.0-py3-none-any.whl
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10.3k/10.3k [00:00<00:00, 22.4kB/s]
Received "500: Internal Server Error" Package upload appears to have failed.  Retry 4 of 5
Uploading django_dynamic_filenames-1.0.0-py3-none-any.whl
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10.3k/10.3k [00:01<00:00, 9.32kB/s]
Received "500: Internal Server Error" Package upload appears to have failed.  Retry 5 of 5
HTTPError: 500 Server Error: Internal Server Error for url: https://upload.pypi.org/legacy/
(env) MONDO-0262:django-dynamic-filenames johannes$ twine --version
twine version 1.11.0 (pkginfo: 1.4.2, requests: 2.19.1, setuptools: 40.2.0,
requests-toolbelt: 0.8.0, tqdm: 4.25.0)
```

See also: https://travis-ci.com/codingjoe/django-dynamic-filenames/jobs/144208217

**My Platform**
Both locally and on travis-ci

```bash
$ twine --version
twine version 1.11.0 (pkginfo: 1.4.2, requests: 2.19.1, setuptools: 40.2.0,
requests-toolbelt: 0.8.0, tqdm: 4.25.0)
```
https://travis-ci.com/codingjoe/django-dynamic-filenames/jobs/144208217

<!--
    Any details about your specific platform:
    * If the problem is in the browser, what browser, version, and OS?
    * If the problem is with a command-line tool, what version of that tool?
    * If the problem is with connecting to PyPI, include some details about
      your network, including SSL/TLS implementation in use, internet service
      provider, and if there are any firewalls or proxies in use.
-->

**Additional context**
<!-- Add any other context, links, etc. about the feature here. -->

That's the package that I am trying to release:
https://github.com/codingjoe/django-dynamic-filenames/tree/1.0.0

The tracktrace from Warehouse:

```
[2018-09-07 15:25:37 +0000] [508] [ERROR] Error handling request /legacy/
Traceback (most recent call last):
  File "/opt/warehouse/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 135, in handle
    self.handle_request(listener, req, client, addr)
  File "/opt/warehouse/lib/python3.6/site-packages/gunicorn/workers/sync.py", line 182, in handle_request
    resp.write(item)
  File "/opt/warehouse/lib/python3.6/site-packages/gunicorn/http/wsgi.py", line 333, in write
    self.send_headers()
  File "/opt/warehouse/lib/python3.6/site-packages/gunicorn/http/wsgi.py", line 329, in send_headers
    util.write(self.sock, util.to_bytestring(header_str, "ascii"))
  File "/opt/warehouse/lib/python3.6/site-packages/gunicorn/util.py", line 507, in to_bytestring
    return value.encode(encoding)
UnicodeEncodeError: 'ascii' codec can't encode character '\xb1' in position 90: ordinal not in range(128)
```

--- a/requirements/deploy.in
+++ b/requirements/deploy.in
@@ -1,1 +1,1 @@
-gunicorn==19.9.0
+gunicorn==20.0.4

--- a/requirements/deploy.txt
+++ b/requirements/deploy.txt
@@ -1,3 +1,8 @@
-gunicorn==19.9.0 \
-    --hash=sha256:aa8e0b40b4157b36a5df5e599f45c9c76d6af43845ba3b3b0efe2c70473c2471 \
-    --hash=sha256:fa2662097c66f920f53f70621c6c58ca4a3c4d3434205e608e121b5b3b71f4f3
+gunicorn==20.0.4 \
+    --hash=sha256:1904bb2b8a43658807108d59c3f3d56c2b6121a701161de0ddf9ad140073c626 \
+    --hash=sha256:cd4a810dd51bf497552cf3f863b575dabd73d6ad6a91075b65936b151cbf4f9c
+
+# The following packages are considered to be unsafe in a requirements file:
+setuptools==42.0.2 \
+    --hash=sha256:c5b372090d7c8709ce79a6a66872a91e518f7d65af97fca78135e1cb10d4b940 \
+    --hash=sha256:c8abd0f3574bc23afd2f6fd2c415ba7d9e097c8a99b845473b0d957ba1e2dac6

--- a/requirements/main.txt
+++ b/requirements/main.txt
@@ -618,6 +618,6 @@ zxcvbn==4.4.28 \
     --hash=sha256:151bd816817e645e9064c354b13544f85137ea3320ca3be1fb6873ea75ef7dc1
 
 # The following packages are considered to be unsafe in a requirements file:
-setuptools==41.6.0 \
-    --hash=sha256:3e8e8505e563631e7cb110d9ad82d135ee866b8146d5efe06e42be07a72db20a \
-    --hash=sha256:6afa61b391dcd16cb8890ec9f66cc4015a8a31a6e1c2b4e0c464514be1a3d722
+setuptools==42.0.2 \
+    --hash=sha256:c5b372090d7c8709ce79a6a66872a91e518f7d65af97fca78135e1cb10d4b940 \
+    --hash=sha256:c8abd0f3574bc23afd2f6fd2c415ba7d9e097c8a99b845473b0d957ba1e2dac6

